### PR TITLE
Add label support introduced in V0.17

### PR DESCRIPTION
**Pull request by Sword-Smith**
Cf.
https://bitcoincore.org/en/releases/0.17.0/#label-and-account-apis-for-wallet,
some methods have been deprecated. To ensure backwards compatibility,
the deprecated methods are not removed in this commit.

The gist of the changes is that the "account" concept is replaced by
the "label" concept which map one-to-one to labels used in the
GUI version of Bitcoin Core. Thus the CLI use of labels is
compatible with GUI use of labels. Previously this was not the case.

See the above link for a description of the new methods.

Added methods:
- GetAddressInfo
- GetAddressByLabel
- GetReceivedByLabel
- ListLabels
- ListReceivedByLabel:
  Like ListReceivedByLabel but the field InvolvesWatchOnly is added. It
  indicates if the returned balance includes tx outputs sent to
  watch-only addresses. This field is null if the reveived amount is 0.
- SetLabel

Changed methods:
- ListUnspent:
  Added a "Label" field, but kept the "Account" field for backwards
  compatibility. Both cases are handled in the associated response
  class' ToString method.

The method arguments named "account" have not been renamed to label.
This may be appropriate to do for v0.18 or if the backwards compatibility
is removed from this library.

### CODE REVIEW

**Comment by Sword-Smith** [comment]
This commit is intended to close Issue #93.

**Comment by moodmosaic** [comment]
Thank you! I will make sure to review this over the weekend. It looks promising from what I see already. :100:

**Comment by moodmosaic** [comment]
@Sword-Smith 

>To ensure backwards compatibility, the deprecated methods are not removed in this commit.

I think, sooner or later, we have to drop support for old(er) versions. As stated in the official docs:

>The `account` can continue to be used in V0.17 by starting bitcoind with the `-deprecatedrpc=accounts` argument, and will be fully removed in V0.18.

File: src/BitcoinLib/Responses/ListUnspentResponse.cs [hunk 1]
```diff
@@ -9,14 +9,21 @@ public class ListUnspentResponse
...
         {
-            return $"Account: {Account}, Address: {Address}, Amount: {Amount}, Confirmations: {Confirmations}";
+            // The Account field/concept is deprecated and will be removed in v0.18, cf.:
+            // https://bitcoincore.org/en/releases/0.17.0/#label-and-account-apis-for-wallet
+            if (!(Account is null))
+            {
+                return $"Account: {Account}, Address: {Address}, Amount: {Amount}, Confirmations: {Confirmations}";
+            }
+            return $"Label: {Label}, Address: {Address}, Amount: {Amount}, Confirmations: {Confirmations}";
         }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/ListUnspentResponse.cs]: 👍

---

On a side note, I think it's time to add a note on the README, something like

>The `account` can continue to be used in V0.17 by starting bitcoind with the `-deprecatedrpc=accounts` argument, and will be fully removed in V0.18.

And point at https://bitcoincore.org/en/releases/0.17.0/#label-and-account-apis-for-wallet

(Just a note to self, you don't *have* to update the README. 😃)

**Comment by moodmosaic** [comment]
@Sword-Smith, thank you! This is now up on NuGet as [BitcoinLib 1.11.0](https://www.nuget.org/packages/BitcoinLib/1.11.0).

**Pull request merged**